src/v/raft/recovery_stm.h

```diff
@@ -58,11 +58,15 @@ class recovery_stm {
       = std::variant<storage::snapshot_reader, on_demand_snapshot_reader>;
     ss::future<> recover();
     ss::future<> do_recover(ss::io_priority_class);
-    ss::future<std::optional<model::record_batch_reader>>
+    ss::future<std::optional<std::tuple<model::record_batch_reader, size_t>>>
```

Comment:
It's kind of unfortunate that we have to plumb "size" around everywhere now in addition to the batch reader itself. I suppose this is because the reader doesn't expose a concept of size, and also because we need to know the size up front at the send/no-send decision point in append entries?

Reply:
Exactly, in general record batch reader is not finite, therefore there is no notion of size in it. I have some ideas around making it a bit easier in future (we can replace the reader with simple `chunked_vector` of batches as this is what we are dealing with in Raft layer in batcher, in requests and along the path)

Reply:
I thought that all of the record batch readers used in raft were secretly in-memory readers. if so, could we just formalize that, and then not have this problem?

Reply:
I was thinking about that. Raft interface doesn't need to accept the record batch reader. We can happily operate on chunked vectors and use reader only when appending to log